README.md

```diff
@@ -1,16 +1,1831 @@
+  "input": {
+    "func": "add_one",
+    "args": [
+      42
+    ]
+  }
+}
+```
+
+# 4 Authorization
+
+An [Authorization] is cryptographically signed data set. It represents an authorization to run [Task]s in that are included in `scope` data set.
+
+## 4.1 Schema
+
+```ipldsch
+type Authorization struct {
+  # Authorization is denoted by the set of links been authorized
+  scope   [&Any] (implicit [])
+  # Scope signed by the invoker
+  s       VarSig
+}
+```
+
+### 4.2 Fields
+
+#### 4.2.1 Authorization Scope
+
+The `scope` field MUST be a set of links been authorized. It SHOULD be encoded as an alphabetically ordered list without duplicates.
+
+If `scope` field is omitted, it is implicitly a an empty list and has no practical use as it authorizes nothing.
+
+### 4.2.2 Signature
+
+The `s` field MUST contain a [Varsig] of the [CBOR] encoded `scope` field.
+
+## 4.3 DAG-JSON Example
+
+```json
+{
+  "scope": [
+    {
+      "/": "bafyreigfusg7tgegda7pxdn7px5vs7wxqk5nybr5ewjht4xas357x6ryia"
+    },
+    {
+      "/": "bafyreid32aunjg6g6buib7bva2sa5ni4c6eaftgykyh6z6rnhcjxh4yk2y"
+    }
+  ],
+  "s": {
+    "/": {
+      "bytes": "7aEDQJBsO/9rkcgQQ9qvSC0E60MdqGjpIHXuc4VbkfjEeYR7iAvAY+0QEN9kAFcwh6/kww4bK/I7xsqLzO5kG0x+2Qs"
+    }
+  }
+}
+```
+
+# 5 Invocation
+
+As [noted in the introduction][lazy-vs-eager], there is a difference between a reference to a function and calling that function. The [Invocation] is an instruction to the [Executor] to perform enclosed [Task]. [Invocation]s are not executable until they have been provided provable authority (in form of UCANs in the `prf` field) and an [Authorization] (in the `auth` field) from the [Invoker].
+
+If `auth` field MUST be set to the [Authorization] (link) which contains `run` in it's scope field. The Invocation of the [Task] with an [Authorization] which does not includes [Task] in the `scope` MUST be considered invalid.
+
+## 5.1 Schema
+
+```ipldsch
+type Invocation struct {
+  v       SemVer
+
+  run     &Task
+  # Receipt of the invocation that caused this invocation
+  cause   optional &Invocation
+  # Task authorization.
+  auth    &Authorization
+
+  meta    {String : any}
+
+  prf     [&UCAN]
+}
+type SemVer string
+```
+
+## 5.2 Fields
+
+### 5.2.1 Task
+
+The `run` field MUST contain a link to the [Task] to be run.
+
+### 5.2.2 Authorization
+
+The `auth` field MUST contain a link to the [Authorization] that authorizes invoked [Task] in the `run` field. Linked [Authorization] MUST contain `run` in it's `scope`.
+
+### 5.2.3 Cause
+
+Some [Task]s may be invoked as an effect caused by another [Task] [Invocation]. Such Invocations SHOULD have `cause` field set to the [Invocation] (link) that caused it. The [Receipt] of the causing [Invocation] (Invocation linked from `cause` field) SHOULD have an `Effect` (the `fx` field) containing cased [Invocation].
+
+## 5.3 DAG-JSON Example
+
+### 5.3.1 Single Invocation
+
+```json
+{
+  "blocks": {
```

Comment:
Any objection to dropping the `blocks` key and using `bafy...semanticLabel` as CIDs for explanitory purposes?

Reply:
e.g.

```js

{
  "bafy...createBlogPost": {
    "with": "https://example.com/blog/posts",
    "do": "crud/create",
    "input": /* ... */
  },
  // ...
}

Reply:
I don't mind it per say, that said I thought it was a good idea to serialize payloads as CARs in DAG-JSON format which is why `blocks` and `roots` are here.

Unless this is really important can we do it as a followup spec improvement ? I'm just bit short of work cycles.

Reply:
> that said I thought it was a good idea to serialize payloads as CARs in DAG-JSON format

I agree, and we're likely to do the same — I'm just trying to make it easy to read, and CAR files aren't a requirement.

> I'm just bit short of work cycles.

Right there with you. Sure, let's update it after the fact, and if either of us (or someone else) really cares, then we can change it later 👍